exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -699,7 +699,7 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
     }
 
     /** Sets the default value for this column in the client side. */
-    fun <T : Any> Column<T>.clientDefault(defaultValue: () -> T): Column<T> = apply {
+    fun <T> Column<T>.clientDefault(defaultValue: () -> T): Column<T> = apply {
```

Comment:
Could you add a small test for that in the `DefaultsTest.kt`, please, that now makes sure that you could do either `clientDefault().nullable()` or `nullable().clientDefault()`?

Reply:
So, these changes are in `exposed-core`. `DefaultsTest.kt` seems to exist in `exposed-java-time` and `exposed-kotlin-datetime`.

Wouldn't a test for this be better suiter for something like `CreateTableTests.kt` in `exposed-tests.test`?

Reply:
Actually, on closer inspection I see that CreateTableTests is for schema generation and `DefaultsTest.kt` in `exposed-java-time` and `exposed-kotlin-datetime` seem to be the only places where defaults are tested. So adding tests there.

Reply:
There we go. The two tests just create tables and checks that column is nullable, has a default value function and that the function returns the expected value. No reading/writing to databases.

If you want I can add read/write tests with this as well, similar to the other tests checking that defaults are written. But there shouldn't be a difference in how these operate compared to other columns with client defaults I think, since they are the right type afterall.

There are 8 failing tests in `exposed-kotlin-datetime`. I verified with the `master` branch that these were failing before as well. Seems to simply be an issue of database truncating microseconds to milliseconds.